### PR TITLE
Arm test

### PR DESCRIPTION
**Pull request by ShiCheng-Lu**
test on arm by flashing and running the test files.

use uart output to determine when to flash and test the next file. (needs uart to get fixed for this to be reliable)

### CODE REVIEW

**Comment by ShiCheng-Lu** [comment]
Still need to make minicom add a \r after new lines

File: projects/leds/test/test_led_2.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include "unity.h"
+
+static const GpioAddress leds[] = {
+  { .port = GPIO_PORT_B, .pin = 5 },   //
+  { .port = GPIO_PORT_B, .pin = 4 },   //
+  { .port = GPIO_PORT_B, .pin = 3 },   //
+  { .port = GPIO_PORT_A, .pin = 15 },  //
+};
+
+TASK(leds_task, TASK_STACK_512) {}
```
Comment (mitchellostler) [inline projects/leds/test/test_led_2.c:18]: What's the purpose of creating this task? Isn't the below code called from the test task?

**Review by mitchellostler — APPROVED** [review]
Looks good, I like that you set up serial to print out all the output to console.

File: scons/test.scons [hunk 2]
```diff
@@ -159,8 +160,48 @@ def get_test_list():
...
+            cmd = 'sudo {}'.format(' '.join(OPENOCD_CFG))
+            print(f"Flashing {test}")
+            subprocess.run(cmd, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            print(f"Starting {test}")
+
+            # wait until test ok or fail
+            while True:
+                line: str = serialData.readline().decode("utf-8")
+                print(line, end='')
+                if line.startswith('OK'):
```
Comment (JarvisWeng) [inline scons/test.scons:198]: So we're reading back UART, if task was successful or not right?

File: scons/test.scons [hunk 3]
```diff
@@ -159,8 +160,48 @@ def get_test_list():
...
+                '-c "stm_flash {}"'.format(test),
+                '-c shutdown'
+            ]
+            cmd = 'sudo {}'.format(' '.join(OPENOCD_CFG))
+            print(f"Flashing {test}")
+            subprocess.run(cmd, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            print(f"Starting {test}")
+
+            # wait until test ok or fail
+            while True:
```
Comment (JarvisWeng) [inline scons/test.scons:195]: Maybe have a timeout here and just assume failed if too long

File: scons/test.scons (hunk 2 above)
Comment (ShiCheng-Lu) [reply scons/test.scons:198]: yes

File: projects/leds/test/test_led_2.c (hunk 1 above)
Comment (ShiCheng-Lu) [reply projects/leds/test/test_led_2.c]: forogt

**Pull request merged**